client/src/components/App.jsx

```diff
@@ -2,27 +2,82 @@ import React from 'react';
 import HeaderLeft from './HeaderLeft.jsx';
 import HeaderRight from './HeaderRight.jsx';
 
-// eslint-disable-next-line react/prefer-stateless-function
 class App extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      fill_me_in : ''
+      detailsModalIsOpen: false,
+      shareModalIsOpen: false,
+      saveModalIsOpen: false
     };
+
+    this.openDetailsModal = this.openDetailsModal.bind(this);
+    this.closeDetailsModal = this.closeDetailsModal.bind(this);
+    this.openShareModal = this.openShareModal.bind(this);
+    this.closeShareModal = this.closeShareModal.bind(this);
+    this.openSaveModal = this.openSaveModal.bind(this);
+    this.closeSaveModal = this.closeSaveModal.bind(this);
+  }
+
+  openDetailsModal() {
```

Comment:
Looks like these functions have a lot of similarity. Might be a good idea to create a general function and invoke with different args.